### DISCUSSION TITLE
Release 1.4.0rc1 - test it now please!

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#8234] [General]
## borgbackup 1.4.0rc1

First release candidate for the new 1.4-maint branch and potentially last chance for pre-release testing (you can help testing by using this **additionally** to your production backups).

borg 1.4 is kind of a refreshed 1.2 with mostly the same features and behavior, but a few bigger changes that could introduce issues and thus were not suitable for just releasing them as 1.2.x, but rather need some testing first.

Especially authors of scripts, wrappers, automations and GUI frontends may want to look at BORG_EXIT_CODES.
Also, users of filesystem ACLs on Linux, FreeBSD and macOS should thoroughly test this, there were major changes in the ACL code.

All stuff listed in the changelog should get practically tested, especially correct behavior with BORG_EXIT_CODES=modern set and in error/warning conditions and backup / restore of ACLs.

Long changelog:
 
https://borgbackup.readthedocs.io/en/1.4.0rc1/changes.html#version-1-4-0rc1-2024-05-26

Short borg 1.4 overview (from a borg 1.2 perspective):

https://www.borgbackup.org/releases/borg-1.4.html

### Installation

If you use pip to install this, use: pip install "borgbackup==1.4.0rc1"

For other installation methods and more details, please see: https://borgbackup.org/

See also the 00_README.txt file in the assets list below for a description of the available "fat binary" downloads.

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/1.4.0rc1'>Release 1.4.0rc1 - test it now please!</a>.</em>

**Comment by yrro** [comment]
Hi - binaries for glibc 2.28-based systems are missing - this version is used by RHEL 8.

**Reply by ThomasWaldmann** [reply]
Guess that is because I removed the broken debian buster based build box, see #8171.

**Reply by ThomasWaldmann** [reply]
BTW, did you try the currently offered fat binaries?

There's a slight chance that it could work if no required symbol is missing.

**Reply by yrro** [reply]
I tried the next version but it looks like the embedded Python needs glibc 2.29.

```
yrro@host:/tmp130$ ./borg-linux-glibc231 
[1473311] Error loading Python lib '/tmp/_MEIbEHDsg/libpython3.11.so.1.0': dlopen: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/_MEIbEHDsg/libpython3.11.so.1.0)
```

RHEL 8 will be supported until 2029, so it's worth considering if you want a supported glibc 2.28 platform past that date (can be used for free with a 'developer subscription', it's less annoying than it sounds), on the other hand RHEL 8 users can probably get by with installing from source, or pip, or building their RPMs, I just ended up using the fat binaries because they were more convenient) :)

**Reply by ThomasWaldmann** [reply]
OK, pity.

Guess I'll choose between 2 options:
- either drop buster as planned (users needing support for older systems can just stay on borg 1.2.x on these systems)
- or fix the buster box and just use it even after it runs out of support (like with borg 1.2.x)

I won't deal with RPM-based sytems as I am not used to them.

**Reply by yrro** [reply]
Just realised I never followed up on this--thanks for restoring the glibc 2.28 builds.

**Comment by lfam** [comment]
I've been testing 1.4.0rc1 within GNU Guix, and it seems to be working fine. I've only exercised the basic functionality: create with LZMA compression, extract, info, and list. But the results seem fine.

**Reply by ThomasWaldmann** [reply]
OK, thanks for testing!

Anyone else also having done some tests?
